megamek/src/megamek/server/Server.java

```diff
@@ -31119,6 +31156,22 @@ private void sendReport() {
      * Send the round report to all connected clients.
      */
     private void sendReport(boolean tacticalGeniusReport) {
+        if (mailer != null) {
+            for (var player: game.getPlayersVector()) {
+                if (!StringUtil.isNullOrEmpty(player.getEmail())) {
```

Comment:
Let's set this to send one email per unique email address in the email address collection. I don't think there's any value in sending multiple report emails to one person.

Additionally, this seems like something that's best spun off into a separate thread, as I don't think the server should be waiting for a potentially lengthy email sending process to complete before moving on.

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Executors.html#newSingleThreadExecutor()

Reply:
> Let's set this to send one email per unique email address in the email address collection. I don't think there's any value in sending multiple report emails to the same address.

This is a little bit difficult to do since for double blind situations, each player needs to get their own copy of the report. In the end, since the email address isn't a remembered preference, if people don't want to receive multiple copies, they simply shouldn't enter their email address more than once.

> Additionally, this seems like something that's best spun off into a separate thread, as I don't think the server should be waiting for a potentially lengthy email sending process to complete before moving on.

Yep, email sending is already handled on a separate thread in EmailService (see commit 7be6ed1c). The executor framework was overkill for this, and wouldn't easily allow sending multiple queued messages over the same transport, so I just went with a minimal solution instead.